docs/book/how-to/steps-pipelines/advanced_features.md

```diff
@@ -303,6 +303,185 @@ The fan-in, fan-out method has the following limitations:
+        
+        if len(completed_runs) == len(run_ids):
+            print("All chunk processing runs completed!")
+            break
+        
+        print(f"Completed: {len(completed_runs)}/{len(run_ids)} runs")
+        time.sleep(10)  # Wait 10 seconds before checking again
+    
+    return run_ids
+
+
+@step
+def aggregate_results(run_ids: List[UUID]) -> dict:
+    """Aggregate results from all chunk processing runs."""
+    client = Client()
+    aggregated_results = {}
+    failed_runs = []
+    
+    for run_id in run_ids:
+        run = client.get_pipeline_run(run_id)
+        
+        # Check if run succeeded
+        if run.status.value == "failed":
+            failed_runs.append({
+                "run_id": str(run_id),
+                "status": run.status.value,
+            })
+            print(f"WARNING: Run {run_id} failed with status {run.status.value}")
+            continue
+        
+        # Extract results from successful runs only
+        if "process_chunk" in run.steps:
+            step_run = run.steps["process_chunk"]
+            # Simple assumption: process_chunk step has one output that we can load
+            chunk_result = step_run.output.load()
+            aggregated_results[str(run_id)] = chunk_result
+
+    
+    # Log summary of results
+    total_runs = len(run_ids)
+    successful_runs = len(aggregated_results)
+    failed_count = len(failed_runs)
+    
+    print(f"Aggregation complete: {successful_runs}/{total_runs} runs successful")
+
+    return {
+        "successful_results": aggregated_results,
+        "failed_runs": failed_runs,
+        "summary": {
+            "total_runs": total_runs,
+            "successful_runs": successful_runs,
+            "failed_runs": failed_count
+        }
+    }
+
+
+@pipeline(enable_cache=False)
+def fan_out_fan_in_pipeline(template_id: Optional[UUID] = None):
+    """Fan-out/fan-in pipeline that orchestrates dynamic chunk processing."""
+    # Load chunks dynamically at runtime
+    chunks = load_relevant_chunks()
+    
+    # Trigger chunk processing runs and wait for completion
+    run_ids = trigger_chunk_processing(chunks, template_id)
+    
+    # Aggregate results from all runs
+    results = aggregate_results(run_ids)
+    
+    return results
+
+
+# Define the chunk processing pipeline that will be triggered
+@step
+def process_chunk(chunk_id: Optional[str] = None) -> dict:
+    """Process a single chunk of data."""
+    # Simulate chunk processing
+    print(f"Processing chunk: {chunk_id}")
+    return {
+        "chunk_id": chunk_id,
+        "processed_items": 100,
+        "status": "completed"
+    }
+
+
+@pipeline
+def chunk_processing_pipeline():
+    """Pipeline that processes a single chunk."""
+    result = process_chunk()
+    return result
+
+
+# Usage example
+if __name__ == "__main__":
+    # First, make sure you run the chunk_processing_pipeline once 
+    #  on a remote orchestrator:
+    # chunk_processing_pipeline()
```

Comment:
Instead of doing this and later calling `Client().create_run_template`, you can simply call `chunk_processing_pipeline.create_run_template(name=...)`

Reply:
Didnt know that! Thats great!